lib/src/user.dart

```diff
@@ -183,6 +200,102 @@ class UserProfile {
   const UserProfile(this._data);
 }
 
+/// A class exposing functionality for users to manage API keys from the client. It is always scoped
+/// to a particular [User] and can only be accessed via [User.apiKeys]
+class ApiKeyClient {
+  final User _user;
+
+  ApiKeyClient._(this._user);
+
+  /// Creates a new API key with the given name. The value of the returned key
+  /// must be persisted as this is the only time it is available.
+  Future<ApiKey> create(String name) async {
+    return realmCore.createApiKey(_user, name);
+  }
+
+  /// Fetches a specific API key by id.
+  Future<ApiKey?> fetch(ObjectId id) {
+    return _handle404(realmCore.fetchApiKey(_user, id));
+  }
+
+  /// Fetches all API keys associated with the user.
+  Future<List<ApiKey>> fetchAll() async {
+    return realmCore.fetchAllApiKeys(_user);
+  }
+
+  /// Deletes a specific API key by id.
+  Future<void> delete(ObjectId objectId) {
+    return _handle404Void(realmCore.deleteApiKey(_user, objectId));
+  }
+
+  /// Disables an API key by id.
+  Future<void> disable(ObjectId objectId) {
+    return _handle404Void(realmCore.disableApiKey(_user, objectId), id: objectId);
+  }
+
+  /// Enables an API key by id.
+  Future<void> enable(ObjectId objectId) {
+    return _handle404Void(realmCore.enableApiKey(_user, objectId), id: objectId);
+  }
+
+  Future<ApiKey?> _handle404(Future<ApiKey> future) async {
+    try {
+      return await future;
+    } on AppException catch (e) {
+      if (e.statusCode == 404) {
+        return null;
+      }
+
+      rethrow;
+    }
+  }
+
+  Future<void> _handle404Void(Future<void> future, {ObjectId? id}) async {
+    try {
+      await future;
+    } on AppException catch (e) {
+      if (e.statusCode == 404) {
+        if (id != null) {
+          throw AppInternal.createException("Failed to execute operation because ApiKey with Id: {id} doesn't exist.", e.linkToServerLogs, 404);
+        }
+
+        // If the id is null, we just ignore the exception
```

Comment:
`delete` doesn't provide an id argument - the idea is that deleting a key that was already deleted is a no-op.

Reply:
then the question might be why do we require it for `enable` and `disable` and not for `fetch/fetchAll` and `delete`. Why not make `enable/disable` no-op on non existing keys? Right now they will throw an exception

Reply:
If you delete something that doesn't exist, then the end result is the same as if it existed - the thing doesn't exist. With `enable`/`disable`, there's an expectation that the thing exists after the call - with `enable`, you should even be able to login with the API key. So in that situation, throwing seems better since silently ignoring the error would mislead the developer into thinking everything is fine.